### DISCUSSION TITLE
Add site specific condition to Sales.

### DISCUSSION THREAD

**Discussion by samuelbirch** [craftcms/commerce#2459] [Ideas]
Can we have the ability to apply sales/discounts to carts on specific sites in a multisite setup please?

**Comment by nfourtythree** [comment]
Hi @samuelbirch

Thank you for the issue. You are able to do this at the moment. 

Discounts have a [order/discount condition formula](https://craftcms.com/docs/commerce/3.x/discounts.html#discount-condition-formula) which uses the order and Twig condition syntax to determine if the discount should be applied. In this you could create a condition formula check the order's `orderSiteId`.

Sales live outside of the realms of orders, this is why currently they do not have a condition formula field like discounts. However, you can use the `Sales::EVENT_BEFORE_MATCH_PURCHASABLE_SALE` event to be able to return the validity of the sale and if it should be applied.

We are trying to avoid adding too many multisite specific features ahead of the multistore update that will come to Commerce in the future.

I will tweak this issue to reflect what is currently missing.

Thanks!
